image/image.go

```diff
@@ -70,28 +75,49 @@ func (i *genericImage) Signatures() ([][]byte, error) {
 }
 
 func (i *genericImage) Inspect() (*types.ImageInspectInfo, error) {
-	// TODO(runcom): unused version param for now, default to docker v2-1
-	m, err := i.getSchema1Manifest()
+	m, err := i.getParsedManifest()
 	if err != nil {
 		return nil, err
 	}
-	ms1, ok := m.(*manifestSchema1)
-	if !ok {
-		return nil, fmt.Errorf("error retrivieng manifest schema1")
-	}
-	v1 := &v1Image{}
-	if err := json.Unmarshal([]byte(ms1.History[0].V1Compatibility), v1); err != nil {
-		return nil, err
+	switch t := m.(type) {
+	case *manifestSchema1:
```

Comment:
… or, perhaps, a `getv1Image` method on the `genericManifest` interface, and keeping the `ImageInspectInfo` construction common in the two cases? Either way, after this type switch is gone, the rest does not matter that much.